src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -18,132 +18,359 @@ should be included in the pipeline documentation.
+| `SphericalHarmonicDegree`   | [Spherical harmonics](#spherical-harmonics)                                                                                                                                                                                                                          | REQUIRED. Integer. Maximal degree of the spherical harmonic basis employed.                                                                                                                                                                                              |
+
+#### Spherical Harmonics bases
+
+-   `MRtrix3`
+
+    -   Antipodally symmetric: all basis functions with odd degree are
+        assumed zero; `AntipodalSymmetry` MUST NOT be set to True.
+
+    -   Functions assumed to be real: conjugate symmetry is assumed, i.e.
+        *Y*(*l*,-*m*) = *Y*(*l*,*m*)\*, where \* denotes the complex
+        conjugate.
 
--   `DSI` :
+    -   Mapping of image volumes to spherical harmonic basis function
+        coefficients:
+
+        | **Volume** | **Coefficient**                   |
+        | ---------- | --------------------------------- |
+        | 0          | *l* = 0, *m* = 0                  |
+        | 1          | *l* = 2, *m* = 2 (imaginary part) |
+        | 2          | *l* = 2, *m* = 1 (imaginary part) |
+        | 3          | *l* = 2, *m* = 0                  |
+        | 4          | *l* = 2, *m* = 1 (real part)      |
+        | 5          | *l* = 2, *m* = 2 (real part)      |
+        | 6          | *l* = 4, *m* = 4 (imaginary part) |
+        | 7          | *l* = 4, *m* = 3 (imaginary part) |
+        | ...        | etc.                              |
+
+    -   Normalisation: ***TODO***
+
+    Relationship between maximal spherical harmonic degree *l<sub>max</sub>*
+    and number of image volumes *N*:
+
+        *N* = ((*l<sub>max</sub>*+1) x (*l<sub>max</sub>*+2)) / 2
+
+        | ***l<sub>max</sub>*** | ***N*** |
+        | --------------------- | ------- |
+        | 0                     | 1       |
+        | 2                     | 6       |
+        | 4                     | 15      |
+        | 6                     | 28      |
+        | 8                     | 45      |
+        | ...                   | etc.    |
+
+    Relationship between maximal degree of *zonal* spherical harmonic
+    function (spherical harmonics function where all *m* != 0 terms are
+    assumed to be zero; used for e.g. response function definition) and
+    number of coefficients *N*:
+
+        *N* = 1 + (*l<sub>max</sub>* / 2)
+
+        | ***l<sub>max</sub>*** | ***N*** |
+        | --------------------- | ------- |
+        | 0                     | 1       |
+        | 2                     | 2       |
+        | 4                     | 3       |
+        | 6                     | 4       |
+        | 8                     | 5       |
+        | ...                   | etc.    |
+
+ -   `Descoteaux`

```

Comment:
As discussed: likely what will be most succinct is a description of how this convention differs from the *MRtrix3* one. Many of the details will be shared between the two.

Reply:
I have the content, I just need to format it correctly. Do you want to get a PR on your PR? Or I just past it in a comment and you update it yourself?

Reply:
While it's possible to make proposed changes to a PR within a comment, I think I'd rather strip it from this PR, merge, and then we can start making individual PRs for individual changes like this. It will also make the relevant diffs far easier to navigate. We shouldn't see this individual PR as the "final" BEP.